geoengine/datasets.py

```diff
@@ -126,18 +131,18 @@ def to_dict(self) -> Dict[str, str]:
 class OgrSourceDatasetTimeType:
     '''A time type specification for OGR dataset definitions'''
 
-    def to_dict(self) -> Dict[str, str]:
+    def to_dict(self) -> Dict[str, Union[str, Dict[str, str]]]:
         pass
```

Comment:
Isn't the method body missing?

Reply:
I assumed this to be sort of an abstract method that is intended to be without any body.
The same situation is here: https://github.com/geo-engine/geoengine-python/blob/b7dce883d39eb0f6edaeecf4df25a632eb09f79b/geoengine/datasets.py#L24-L25
and here: https://github.com/geo-engine/geoengine-python/blob/b7dce883d39eb0f6edaeecf4df25a632eb09f79b/geoengine/datasets.py#L76-L77

Reply:
Is it worth using annotations from here, then?
https://docs.python.org/3/library/abc.html

Reply:
I think this would be a good addition. 
A simple integration with [52f3cec](https://github.com/geo-engine/geoengine-python/pull/70/commits/52f3cec883e67a03700efe7f8dc4db0beaf55494)